File: plugins/cc-bridge/scripts/readers/env.py
"""Read the `env` key from Claude Code `.claude/settings*.json` files."""

from __future__ import annotations

import json
from pathlib import Path


def _load_env(path: Path) -> dict[str, str]:
    if not path.is_file():
        return {}
    try:
        data = json.loads(path.read_text(encoding="utf-8"))
    except (json.JSONDecodeError, OSError):
        return {}
    env = data.get("env") if isinstance(data, dict) else None
    if not isinstance(env, dict):
        return {}
    return {str(k): str(v) for k, v in env.items()}


def read_env(project_root: Path) -> dict[str, str]:
    claude_dir = project_root / ".claude"
    project = _load_env(claude_dir / "settings.json")
    local = _load_env(claude_dir / "settings.local.json")
    return {**project, **local}

File: plugins/cc-codex-bridge/src/readers/env.ts
import fs from "node:fs";
import path from "node:path";

export type EnvMap = Record<string, string>;

function loadEnv(filePath: string): EnvMap {
  if (!isFile(filePath)) return {};
  let raw: string;
  try {
    raw = fs.readFileSync(filePath, "utf-8");
  } catch {
    return {};
  }
  let data: unknown;
  try {
    data = JSON.parse(raw);
  } catch {
    return {};
  }
  if (!isPlainObject(data)) return {};
  const env = (data as Record<string, unknown>).env;
  if (!isPlainObject(env)) return {};
  const out: EnvMap = {};
  for (const [k, v] of Object.entries(env)) {
    out[String(k)] = String(v);
  }
  return out;
}

export function readEnv(projectRoot: string): EnvMap {
  const claudeDir = path.join(projectRoot, ".claude");
  const project = loadEnv(path.join(claudeDir, "settings.json"));
  const local = loadEnv(path.join(claudeDir, "settings.local.json"));
  return { ...project, ...local };
}

function isFile(p: string): boolean {
  try {
    return fs.statSync(p).isFile();
  } catch {
    return false;
  }
}

function isPlainObject(value: unknown): value is Record<string, unknown> {
  return typeof value === "object" && value !== null && !Array.isArray(value);
}
